llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp

```diff
@@ -161,6 +161,8 @@ bool matchREV(MachineInstr &MI, MachineRegisterInfo &MRI,
   Register Dst = MI.getOperand(0).getReg();
   Register Src = MI.getOperand(1).getReg();
   LLT Ty = MRI.getType(Dst);
+  if (Ty.isScalableVector())
```

Comment:
Why are these needed? I don't believe there should be any shuffles on scalable types.

I think that goes for most of the changes in this file. Are the all speculative changes in reality are not necessary?

Reply:
I got asserts and went over almost all of them. The stack trace did not tell me where the crash was.

Reply:
I got asserts and went over almost all of them. The stack trace did not tell me where the crash was.